src/main/java/org/folio/rest/impl/RequestExpiryImpl.java

```diff
@@ -30,10 +30,18 @@ public void expireRequests(Map<String, String> okapiHeaders,
 
     Vertx vertx = context.owner();
     new ConfigurationClient(vertx, okapiHeaders).getTlrSettings()
-    .compose(tlrSettings -> createRequestExpirationService(okapiHeaders, vertx, tlrSettings)
-        .doRequestExpiration())
-    .onSuccess(x -> handler.handle(succeededFuture(respond204())))
-    .onFailure(e -> handler.handle(succeededFuture(respond500WithTextPlain(e.getMessage()))));
+      .onSuccess(tlrSettings -> doExpiration(tlrSettings, handler, okapiHeaders, vertx))
+      .onFailure(t -> doExpiration(new TlrSettingsConfiguration(false, false, null, null, null), 
```

Comment:
This handler will user default settings in case of any error. And I think we should only do that if we receive an empty but successful response from mod-configuration (i.e. when settings are not present in its DB). If the settings we fetched are invalid, or if okapi returns an HTTP-error, or in any other case where we fail to obtain a valid set of TLR settings, we should fail as usual.

Maybe we should try and handle missing settings on `ConfigurationClient`'s side? By adding a new method, something like this:
```
public Future<TlrSettingsConfiguration> getTlrSettingsOrDefault() {
   return getTlrSettings().recover(throwable -> {
        if (settings were not found) {
          return default settings
        } else {
          rethrow throwable 
      }
     );
```

Reply:
As I noted in my comments above, I tried that solution and found it not viable.  Other processes in this module expect to receive errors if the TLR settings are empty-there are tests for that condition in the automated tests.  So it needs to continue to issue an empty object.

HTTP errors are checked for and caught elsewhere in this chain.  This condition only triggers if the call to the configuration settings returns a 200 but the object itself is either empty or otherwise unreadable.

Reply:
> Other processes in this module expect to receive an empty object if the settings have not been initialized
I see that the only other process fetching TLR-configuration is TLR-feature-toggle job, and it's expecting an error from `ConfigurationClient` in case if settings were not initialized, not an empty object. The solution I offered above will not affect this behavior, as it introduces a new method which will only be invoked by request expiration service. TLR-feature-toggle job will still call the original `ConfigurationClient#getTlrSettings`.

Reply:
Ah, I see-I didn't get that you were proposing a completely new fetch method that only the expiry code would use.  I can try that and see if it works.